### ISSUE TITLE
MSVC "makefile.bat" fails if certain external tool dependencies are not met

### ISSUE THREAD

**Issue by ghost** [hercules-390/hyperion#61]
MSVC `makefile.bat` fails if certain external tool dependencies are not met

When the Hercules repository is a git repository for example, makefile.bat's "set_VERSION" function attempts to construct a version with the "V4" value set to the commit count by redirecting the output of the git log --pretty=format command into the "wc.exe" tool to perform a word count on git log's results:

```
for /f "usebackq" %%a in (`%git% log --pretty^=format:'' ^| wc -l`) do set "V4=%%a"`
```

If **wc.exe** does not exist anywhere on the system however, makefile.bat encounters an error thereby causing the build to fail.

This is not good.

Successfully building Hercules should not rely on any external tools unless such tools are absolutely necessary, and if they are, such dependencies should be _clearly_ documented as such with information on where to them.

The same situation arises with newer versions of Visual Studio too, which do not ship with a **win32.mak** file: the build aborts due to the missing win32.mak file (`!INCLUDE <win32.mak>` statement in member `makefile-dllmod.msvc`).

`makefile.bat` needs to detect and more gracefully handle such situations to prevent the build from aborting and the build documentation needs to be updated to at least mention such (optional and/or required) dependencies.

**Comment by Fish-Git2** [comment]
Fix committed for "wc.exe" dependency.

The win32.mak dependency issue with later versions of Visual Studio still remains.

I tried installing VS2015 Community Edition on my Win7 virtual machine but was not able to determine a proper workaround/resolution. Maybe someone else can figure it out because I've given up. The ultimate resolution *may* be to simply move Hyperion's build system to "cmake".

Until this issue is *completely* resolved however, I'm going to keep this issue open and simply mark it as "stalled".

**Comment by Fish-Git** [comment]
FYI: makefile `<win32.mak>` is no longer an issue as long as one installs Windows 7.1a SDK and either: a) adds the path to its "include" directory (where `<win32.mak>` lives) to your INCLUDE environment variable, or b) adds it to the project's Include Directories property (n later versions of Visual Studio, such as VS2015).

Note: Windows 7.1a SDK is automatically installed as part of the VS2015 Community Edition installation as long as one selects "Windows XP support" (in C++ section) as one of the components to be installed.

**Comment by srorso** [comment]
Several commits have addressed this issue, most notably 992d5c6, which moved version processing into a new _dynamic_version.cmd, and e0d6d7, which added checks for the presence of win32.mak and appropriate diagnostics in its absence.
